### PR TITLE
KOGITO-764: Create a Guided Tour for first time users on DMN editor

### PR DESCRIPTION
**Pull request by karreiro**
See https://issues.redhat.com/browse/KOGITO-764

Screen Shot 2020-06-17 at 10 09 17

---

Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/988
- https://github.com/kiegroup/kie-wb-common/pull/3336
- https://github.com/kiegroup/kogito-tooling/pull/182

---

Assumptions:
- This PR does not include i18n for any tour. It will be separately implemented by https://issues.redhat.com/browse/KOGITO-2471;
- This PR does not include the final version of the tour, and the steps are a mix of the preliminary input provided by Liz and the Learn DMN in 15 minutes (as accordingly with @ederign).

### CODE REVIEW

File: appformer-kogito-bridge/src/main/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourBridge.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import javax.inject.Inject;
+
+import org.appformer.kogito.bridge.client.guided.tour.GuidedTourCustomSelectorPositionProvider.PositionProviderFunction;
+import org.appformer.kogito.bridge.client.guided.tour.observers.GlobalHTMLObserver;
+import org.appformer.kogito.bridge.client.guided.tour.service.GuidedTourService;
+import org.appformer.kogito.bridge.client.guided.tour.service.api.Tutorial;
+import org.appformer.kogito.bridge.client.guided.tour.service.api.UserInteraction;
+
+@ApplicationScoped
+public class GuidedTourBridge {
```
Comment (jomarko) [inline appformer-kogito-bridge/src/main/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourBridge.java:33]: Could we please add javadoc to all new classes and their public methods?

File: appformer-kogito-bridge/src/test/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourBridgeTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        final GuidedTourCustomSelectorPositionProvider positionProvider = mock(GuidedTourCustomSelectorPositionProvider.class);
+        final PositionProviderFunction positionProviderFunction = mock(PositionProviderFunction.class);
+        final String type = "type";
+
+        doReturn(positionProvider).when(bridge).getPositionProviderInstance();
+
+        bridge.registerPositionProvider(type, positionProviderFunction);
+
+        verify(bridge).registerPositionProvider(type, positionProviderFunction);
+    }
```
Comment (jomarko) [inline appformer-kogito-bridge/src/test/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourBridgeTest.java:124]: This test is not correct

File: appformer-kogito-bridge/src/test/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourCustomSelectorPositionProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        verify(console).warn("[Guided Tour - Position Provider] The position provider could not be found: TEST_PROVIDER_3");
+        assertNotNull(position);
+    }
+
+    @Test
+    public void testGetPositionWhenSelectorIsNull() {
+        final Rect position = positionProvider.getPosition(null);
+
+        verify(console).warn("[Guided Tour - Position Provider] Invalid custom query selector: null");
+        assertNotNull(position);
```
Comment (jomarko) [inline appformer-kogito-bridge/src/test/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourCustomSelectorPositionProviderTest.java:86]: Shouldn't we check if `position` is `Rect.NONE`?

**Comment by jomarko** [comment]
mvn reports 
```
[WARNING] Unused declared dependencies found:
[WARNING]    com.google.inject:guice:jar:4.0:test
```

Could you please double check it is not caused by this PR?

File: appformer-kogito-bridge/src/main/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourBridge.java (hunk 1 above)
Comment (karreiro) [reply appformer-kogito-bridge/src/main/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourBridge.java:36]: Java docs were added.

File: appformer-kogito-bridge/src/test/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourBridgeTest.java (hunk 2 above)
Comment (karreiro) [reply appformer-kogito-bridge/src/test/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourBridgeTest.java]: Well spotted! Thank you!

File: appformer-kogito-bridge/src/test/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourCustomSelectorPositionProviderTest.java (hunk 3 above)
Comment (karreiro) [reply appformer-kogito-bridge/src/test/java/org/appformer/kogito/bridge/client/guided/tour/GuidedTourCustomSelectorPositionProviderTest.java]: I've improved tests on this class. Unfortunately, `Rect` is a native type so we have some assertion limitations, but I believe this test class is good enough now.

**Comment by karreiro** [comment]
Thank you for the review @jomarko. Changes applied.

I've double checked and I don't believe this PR inserts anything related to this `com.google.inject:guice warning`.

**Pull request merged**